src/components/MapboxMap.tsx

```diff
@@ -0,0 +1,157 @@
+  PinComponent?: PinComponent<T>,
+  /** A function to derive a result's coordinate for the corresponding marker's location on the map. */
+  getCoordinate?: CoordinateGetter<T>,
+  /** A function which is called when user drag the map. */
+  onDrag?: (center: LngLat, bounds: LngLatBounds) => void
+}
+
+/**
+ * A component that renders a map with markers to show result locations using Mapbox GL.
+ *
+ * @remarks
+ * For the map to work properly, be sure to include Mapbox GL stylesheet in the application.
+ *
+ * @param props - {@link MapboxMapProps}
+ * @returns A React element conatining a Mapbox Map
+ *
+ * @public
+ */
+export function MapboxMap<T>({
+  mapboxAccessToken,
+  mapboxOptions,
+  PinComponent,
+  getCoordinate = getDefaultCoordinate,
+  onDrag
+}: MapboxMapProps<T>) {
+  useEffect(() => {
+    mapboxgl.accessToken = mapboxAccessToken;
+  }, [mapboxAccessToken]);
+
+  const mapContainer = useRef<HTMLDivElement>(null);
+  const map = useRef<Map | null>(null);
+  const markers = useRef<Marker[]>([]);
+
+  const locationResults = useSearchState(state => state.vertical.results) as Result<T>[];
+  const hasLocationResults = locationResults && locationResults.length !== 0;
+  const onDragDebounced = useDebouncedFunction(onDrag, 100);
+
+  useEffect(() => {
+    if (mapContainer.current && !map.current) {
+      const options: MapboxOptions = {
+        container: mapContainer.current,
+        style: 'mapbox://styles/mapbox/streets-v11',
+        center: [-73.935242, 40.730610],
+        zoom: 9,
+        ...mapboxOptions
+      };
+      map.current = new Map(options);
+      const mapbox = map.current;
+      mapbox.resize();
+      if (onDragDebounced) {
+        mapbox.on('drag', () => {
+          onDragDebounced(mapbox.getCenter(), mapbox.getBounds());
+        });
+      }
+    }
+  }, [mapboxOptions, onDragDebounced]);
+
+  useEffect(() => {
+    markers.current.forEach(marker => marker.remove());
+    markers.current = [];
+
+    if (map.current && hasLocationResults) {
+      const bounds = new LngLatBounds();
+
+      for (const [index, result] of locationResults.entries()) {
+        const markerLocation = getCoordinate(result);
+        if (markerLocation) {
+          const { latitude, longitude } = markerLocation;
+          const el = document.createElement('div');
+          PinComponent && ReactDOM.render(<PinComponent result={result} index={index} />, el);
+          const markerOptions: MarkerOptions = {
+            element: PinComponent ? el : undefined
+          };
+          const marker = new Marker(markerOptions)
+            .setLngLat({ lat: latitude, lng: longitude })
+            .addTo(map.current);
+          markers.current = [marker, ...markers.current];
+          bounds.extend([longitude, latitude]);
+        }
+      }
+
+      if (!bounds.isEmpty()){
+        map.current.fitBounds(bounds, {
+          padding: { top: 50, bottom: 25, left: 50, right: 50 },
```

Comment:
out of curiosity, is there a reason less padding is needed at the bottom?

Reply:
don't think so, I can make it all 50